dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitorTest.kt

```diff
@@ -1528,6 +1532,103 @@ internal class DatadogRumMonitorTest {
+                    override fun get(timeout: Long, unit: TimeUnit?): Any =
+                        error("Not supposed to be called")
+                }
+            }
+        }
+
+        testedMonitor = DatadogRumMonitor(
+            fakeApplicationId,
+            mockSdkCore,
+            fakeSamplingRate,
+            fakeBackgroundTrackingEnabled,
+            fakeTrackFrustrations,
+            mockWriter,
+            mockHandler,
+            mockTelemetryEventHandler,
+            mockDetector,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            mockSessionListener,
+            mockContextProvider,
+            executorService = mockExecutorService
+        )
+
+        var isMethodOccupied = false
+        val mockRootScope = mock<RumScope>().apply {
+            whenever(handleEvent(any(), any())) doAnswer {
+                if (isMethodOccupied) {
+                    throw IllegalStateException(
+                        "Only one thread should" +
+                            " be allowed to enter rootScope at the time."
+                    )
+                }
+                isMethodOccupied = true
+                Thread.sleep(100)
+                isMethodOccupied = false
+                null
+            }
+        }
+        testedMonitor.rootScope = mockRootScope
+        val executor = Executors.newFixedThreadPool(10)
+        val futures = mutableListOf<Future<*>>()
+
+        // When
+        repeat(10) {
+            futures += executor.submit {
+                // we are not going to generate all set of the events, only AddError + fatal
+                // which has a special handling + few simple others
+                val event = forge.anElementFrom(
+                    RumRawEvent.AddError(
+                        message = forge.anAlphaNumericalString(),
+                        source = forge.aValueFrom(RumErrorSource::class.java),
+                        isFatal = true,
+                        throwable = forge.aThrowable(),
+                        stacktrace = forge.anAlphaNumericalString(),
+                        attributes = emptyMap()
+                    ),
+                    RumRawEvent.StartAction(
+                        type = forge.aValueFrom(RumActionType::class.java),
+                        name = forge.anAlphaNumericalString(),
+                        waitForStop = forge.aBool(),
+                        attributes = emptyMap()
+                    ),
+                    RumRawEvent.StartView(
+                        key = Any(),
+                        name = forge.anAlphaNumericalString(),
+                        attributes = emptyMap()
+                    )
+                )
+                testedMonitor.handleEvent(event)
+            }
+        }
+
+        // Then
+        assertDoesNotThrow {
+            futures.forEach {
+                // none of these should throw
+                it.get()
```

Comment:
normally this should throw right if in the implementation above you are overriding the `get` method to throw an error.

Reply:
Oh it is not the same method, I think you should add some explanation here :) it is not at all obvious